### PR TITLE
CASMINST-4669 document generic rerun strategy for upgrade failures

### PR DESCRIPTION
**Pull request by dzou-hpe**
## Summary and Scope

document generic rerun strategy for upgrade failures

## Issues and Related PRs

_List and characterize relationship to Jira/Github issues and other pull requests. Be sure to list dependencies._

* Resolves CASMINST-4669

## Testing

_List the environments in which these changes were tested._

### Tested on:

  * `<development system>`
  * Local development environment
  * Virtual Shasta

### Test description:

_How were the changes tested and success verified? If schema changes were part of this change, how were those handled in your upgrade/downgrade testing?_

- Were the install/upgrade-based validation checks/tests run (goss tests/install-validation doc)?
- Were continuous integration tests run? If not, why?
- Was upgrade tested? If not, why?
- Was downgrade tested? If not, why?
- Were new tests (or test issues/Jiras) created for this change?

## Risks and Mitigations

_Are there known issues with these changes? Any other special considerations?_

## Pull Request Checklist

- [ ] Version number(s) incremented, if applicable
- [ ] Copyrights updated
- [ ] License file intact
- [ ] Target branch correct
- [ ] CHANGELOG.md updated
- [ ] Testing is appropriate and complete, if applicable
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

**Review by kimjensen-hpe — APPROVED** [review]

File: upgrade/1.2/README.md [hunk 1]
```diff
@@ -35,6 +35,10 @@ For more information about modifying `customizations.yaml` and tuning for specif
 
 ## Relevant troubleshooting links for upgrade-related issues
 
+- General Upgrade troubleshooting
+
+  If execution of the upgrade procedures fails, it is safe to just rerun the failed script. If a rerun still fails, wait for 10s and rerun it again. If the issue persists, then refer to below troubleshooting procedures.
```
Comment (kimjensen-hpe) [inline upgrade/1.2/README.md:40]: might be better to say "procedure fails" or "procedures fail" depending on the intended meaning

**Review by bklei — APPROVED** [review]

**Review by rgross727 — APPROVED** [review]

File: upgrade/1.2/README.md [hunk 2]
```diff
@@ -35,6 +35,10 @@ For more information about modifying `customizations.yaml` and tuning for specif
 
 ## Relevant troubleshooting links for upgrade-related issues
 
+- General Upgrade troubleshooting
```
Comment (nrockershousen) [inline upgrade/1.2/README.md:38]:
```suggestion
- General upgrade troubleshooting
```

File: upgrade/1.2/README.md [hunk 3]
```diff
@@ -35,6 +35,10 @@ For more information about modifying `customizations.yaml` and tuning for specif
...
+
+  If execution of the upgrade procedure fails, it is safe to just rerun the failed script. If a rerun still fails, wait for 10s and rerun it again. If the issue persists, then refer to below troubleshooting procedures.
```
Comment (nrockershousen) [inline upgrade/1.2/README.md:40]:
```suggestion
  If the execution of the upgrade procedure fails, it is safe to rerun the failed script. If a rerun still fails, wait for 10s and rerun it again. If the issue persists, then refer to the below troubleshooting procedures.
```

**Comment by dzou-hpe** [comment]
> This also needs a main branch PR

no, we want to handle this error in 1.3 and not asking installer to "rerun would fix it"

File: upgrade/1.2/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (dzou-hpe) [reply upgrade/1.2/README.md]: applied suggestion

**Review by jasonrouault — APPROVED** [review]

**Review by mharding-hpe — APPROVED** [review]

**Review by mharding-hpe — APPROVED** [review]

**Pull request merged**